src/Gaffer/PlugAlgo.cpp

```diff
@@ -717,11 +717,40 @@ Plug *promoteWithName( Plug *plug, const InternedString &name, Plug *parent, con
 
 	Node *externalNode = ::externalNode( plug );
 	const bool dynamic = runTimeCast<Box>( externalNode ) || parent == externalNode->userPlug();
-	// We use `persistent = dynamic` so that `promoteWithName()` can be used in
-	// constructors for custom nodes, to promote a plug from an internal
-	// network. In this case, we don't want the metadata to be serialised with
-	// the node, as it will be recreated upon construction anyway.
-	MetadataAlgo::copy( plug, externalPlug.get(), excludeMetadata, /* persistentOnly = */ false, /* persistent = */ dynamic );
+
+	MetadataAlgo::copyIf(
+		plug, externalPlug.get(),
+		[&excludeMetadata]( const GraphComponent *from, const GraphComponent *to, InternedString name ) {
+			if( StringAlgo::matchMultiple( name.string(), excludeMetadata ) )
+			{
+				/// \todo Remove `excludeMetadata` and rely on registered exclusions only. An obstacle
+				/// to doing this is making it easy to exclude `layout:*` without lots and lots of
```

Comment:
`<name>:promotable` -> `plugAlgo:promote:<StringAlgo::match>`?

Reply:
Summary of offline discussion :

- Would be nice to get one step closer to removing the `excludeMetadata` argument.
- The above suggestion would be used as `( "plugAlgo:promote:layout:*", false ), ( "plugAlgo:promote:preset:*, false ), "plugAlgo:promote:presetNames", false ), ( "plugAlgo:promote:presetValues", false )`.
    - Pros :
        - Supports wildcards, so we can deal with `layout:*` elegantly.
        - Can always exclude another thing without needing to know what else is excluded already.
    - Cons :
        - Quadratic scaling if excluding all metadata explicitly without wildcards (N checks against N items). Probably more a theoretical concern?
        - A bit odd to have wildcards in they key?
        - How to resolve multiple matches with different values?
- I'd been wondering about a separate single `"plugAlgo:promote:exclude"` key for use in addition to `<name>:promotable`.
    - Pros : Can deal with `layout:*`. Doesn't have the same scaling worries.
    - Cons : Can't be edited effectively without knowing what the previous value is. Basically it'll get hardcoded at `layout:*` and won't be usable for anything else.
- Will sleep on it.